apps/ex_nvr_web/lib/ex_nvr_web/live/device_live.ex

```diff
@@ -11,31 +14,39 @@ defmodule ExNVRWeb.DeviceLive do
 
   def mount(%{"id" => "new"}, _session, socket) do
     changeset = Devices.change_device_creation(%Device{})
-
-    {:ok,
-     socket
-     |> assign(
-       device: %Device{},
-       disks_data: get_disks_data(),
-       device_form: to_form(changeset),
-       device_type: "ip"
-     )
-     |> allow_upload(:file_to_upload,
-       accept: ~w(video/mp4),
-       max_file_size: 1_000_000_000
-     )}
+    role = socket.assigns.current_user.role
+
+    if is_authorized?(role, :device, :create),
```

Comment:
Look at `live_session` for how to forbid access to this pages. You put `onvif_discovery` and `device` page there and add new authorization function on mounting. (an example can be found on the `router.ex` file)